Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/RdbmsCentralRepo.java

```diff
@@ -2543,88 +2547,55 @@ public void processSelectClause(String selectClause, InstanceTableCallback insta
-        }
-
-        sqlSb.append(selectSQL);
-
-        try (Connection conn = connect();
-                PreparedStatement preparedStatement = conn.prepareStatement(sqlSb.toString());
-                ResultSet resultSet = preparedStatement.executeQuery();) {
+        ResultSet resultSet = null;
+        try ( Connection conn = connect();)   {
+             PreparedStatement preparedStatement = conn.prepareStatement(sql);
+             
+            // fill in the params
+            if (params != null) {
+                int paramIndex = 1;
+                for (Object param : params) {
+                    preparedStatement.setObject(paramIndex, param);
+                    paramIndex += 1;
+                }
+            }
+            // execute query
+            resultSet = preparedStatement.executeQuery();
+           
             queryCallback.process(resultSet);
         } catch (SQLException ex) {
-            throw new CentralRepoException(String.format("Error running SQL %s, exception = %s", selectSQL, ex.getMessage()), ex);
-        }
-    }
-    
-    @Override
-    public void executeUpdateSQL(String updateSQL) throws CentralRepoException {
-
-        if (updateSQL == null) {
-            throw new CentralRepoException("Update SQL is null");
-        }
-
-        StringBuilder sqlSb = new StringBuilder(QUERY_STR_MAX_LEN);
-        if (updateSQL.trim().toUpperCase().startsWith("UPDATE") == false) {
-            sqlSb.append("UPDATE ");
-        }
-        
-        sqlSb.append(updateSQL);
-        
-        try (Connection conn = connect();
-                PreparedStatement preparedStatement = conn.prepareStatement(sqlSb.toString());) {
-            preparedStatement.executeUpdate();
-        } catch (SQLException ex) {
-            throw new CentralRepoException(String.format("Error running SQL %s, exception = %s", updateSQL, ex.getMessage()), ex);
-        }
-    }
-    
-    @Override
-    public void executeDeleteSQL(String deleteSQL) throws CentralRepoException {
-        
-        if (deleteSQL == null) {
-            throw new CentralRepoException("Delete SQL is null");
-        }
-
-        StringBuilder sqlSb = new StringBuilder(QUERY_STR_MAX_LEN);
-        if (deleteSQL.trim().toUpperCase().startsWith("DELETE") == false) {
-            sqlSb.append("DELETE ");
-        }
-
-        sqlSb.append(deleteSQL);
-
-        try (Connection conn = connect();
-                PreparedStatement preparedStatement = conn.prepareStatement(sqlSb.toString());) {
-            preparedStatement.executeUpdate();
-        } catch (SQLException ex) {
-            throw new CentralRepoException(String.format("Error running SQL %s, exception = %s", deleteSQL, ex.getMessage()), ex);
+            throw new CentralRepoException(String.format("Error executing prepared statement for SQL query %s", sql), ex);
+        } finally {
+            if (resultSet != null) {
+                CentralRepoDbUtil.closeResultSet(resultSet);
+            }
```

Comment:
@raman-bt and @gdicristofaro, I think we can hadnle closing of the ResultSet more cleanly using another try with resources.